raiden/tests/integration/long_running/test_integration_events.py

```diff
@@ -431,7 +435,7 @@ def test_secret_revealed_on_chain(
         secret=secret,
     )
 
-    with gevent.Timeout(10):
+    with watch_for_unlock_failures(*raiden_chain), gevent.Timeout(10):
```

Comment:
Currently `watch_for_unlock_failures` check that no bad event among all events in the raiden DB. There are two possible improvements I see here:
- Wrap the whole test with `watch_for_unlock_failures` and have everything checked
- Change `watch_for_unlock_failures` to store the number of events as the `offset` parameter for each app  during initialization and then use it when getting events.

I'm leaning towards option 2 as it prevents false positives and allows for tests with intentional unlock failures.

Reply:
I've been wondering about how to improve `watch_for_unlock_failures` too. The offset parameter is a good idea, I will add it.